File: hedera-mirror-rest/model/entity.js
// SPDX-License-Identifier: Apache-2.0

import _ from 'lodash';

class Entity {
  static historyTableName = 'entity_history';
  static tableAlias = 'e';
  static tableName = 'entity';
  static ALIAS = 'alias';
  static AUTO_RENEW_ACCOUNT_ID = 'auto_renew_account_id';
  static AUTO_RENEW_PERIOD = 'auto_renew_period';
  static CREATED_TIMESTAMP = 'created_timestamp';
  static DELETED = 'deleted';
  static ETHEREUM_NONCE = 'ethereum_nonce';
  static EVM_ADDRESS = 'evm_address';
  static EXPIRATION_TIMESTAMP = 'expiration_timestamp';
  static ID = 'id';
  static KEY = 'key';
  static MAX_AUTOMATIC_TOKEN_ASSOCIATIONS = 'max_automatic_token_associations';
  static MEMO = 'memo';
  static NUM = 'num';
  static OBTAINER_ID = 'obtainer_id';
  static PERMANENT_REMOVAL = 'permanent_removal';
  static PROXY_ACCOUNT_ID = 'proxy_account_id';
  static PUBLIC_KEY = 'public_key';
  static REALM = 'realm';
  static RECEIVER_SIG_REQUIRED = 'receiver_sig_required';
  static SHARD = 'shard';
  static SUBMIT_KEY = 'submit_key';
  static TIMESTAMP_RANGE = 'timestamp_range';
  static TYPE = 'type';

  /**
   * Parses entity table columns into object
   */
  constructor(entity) {
    Object.assign(
      this,
      _.mapKeys(entity, (v, k) => _.camelCase(k))
    );
  }

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default Entity;

File: rosetta/app/persistence/domain/entity.go
// SPDX-License-Identifier: Apache-2.0

package domain

import "github.com/jackc/pgtype"

const (
	EntityTypeAccount = "ACCOUNT"

	entityTableName        = "entity"
	entityHistoryTableName = "entity_history"
)

type Entity struct {
	Alias                         []byte
	AutoRenewAccountId            *EntityId
	AutoRenewPeriod               *int64
	CreatedTimestamp              *int64
	Deleted                       *bool
	ExpirationTimestamp           *int64
	Id                            EntityId
	Key                           []byte
	MaxAutomaticTokenAssociations *int32
	Memo                          string
	Num                           int64
	PublicKey                     *string
	ProxyAccountId                *EntityId
	Realm                         int64
	ReceiverSigRequired           *bool
	Shard                         int64
	TimestampRange                pgtype.Int8range
	Type                          string
}

func (e Entity) GetModifiedTimestamp() int64 {
	return e.TimestampRange.Lower.Int
}

func (Entity) TableName() string {
	return entityTableName
}

func (Entity) HistoryTableName() string {
	return entityHistoryTableName
}
